exporter/splunkhecexporter/buffer.go

```diff
@@ -166,44 +125,34 @@ func (c *cancellableGzipWriter) Empty() bool {
 	return c.rawLen == 0
 }
 
-// bufferStatePool is a pool of bufferState objects.
-type bufferStatePool struct {
+// bufferPool is a pool of buffer objects.
+type bufferPool struct {
 	pool *sync.Pool
 }
 
-// get returns a bufferState from the pool.
-func (p bufferStatePool) get() *bufferState {
-	return p.pool.Get().(*bufferState)
+func (p bufferPool) get() buffer {
+	return p.pool.Get().(buffer)
 }
 
-// put returns a bufferState to the pool.
-func (p bufferStatePool) put(bf *bufferState) {
+func (p bufferPool) put(bf buffer) {
 	p.pool.Put(bf)
 }
 
-const initBufferCap = 512
-
-func newBufferStatePool(bufCap uint, compressionEnabled bool) bufferStatePool {
-	return bufferStatePool{
+func newBufferPool(bufCap uint, compressionEnabled bool) bufferPool {
+	return bufferPool{
 		&sync.Pool{
 			New: func() interface{} {
-				innerBuffer := bytes.NewBuffer(make([]byte, 0, initBufferCap))
-				var buf buffer
+				innerBuffer := &bytes.Buffer{}
 				if compressionEnabled {
-					buf = &cancellableGzipWriter{
+					return &cancellableGzipWriter{
 						innerBuffer: innerBuffer,
-						innerWriter: gzip.NewWriter(buf),
-						maxCapacity: bufCap,
-					}
-				} else {
-					buf = &cancellableBytesWriter{
-						innerWriter: innerBuffer,
+						innerWriter: gzip.NewWriter(innerBuffer),
 						maxCapacity: bufCap,
 					}
 				}
-				return &bufferState{
-					buf:        buf,
-					jsonStream: jsoniter.NewStream(jsoniter.ConfigDefault, nil, initBufferCap),
```

Comment:
you changed the jsoniter config with this change - I can't remember if there is a reason for how it was set, just pointing that out. If tests pass, I guess we're ok.

Reply:
Oh, you're right I changed it to `jsoniter.ConfigFastest` unintentionally. Let me revert as it's not related

Reply:
You still have my approval either way. Since you're reverting that change, maybe worth updating the benchmark numbers real quick as well. I guess you'd want to also run a separate PR with the jsoniter changes.

Reply:
Yeah, the benchmarks appeared to be not a good without with the same jsoniter config. Maybe because of moving the jsoniter stream pool. Let me look into it

Reply:
Ok, I changed it to fetch the stream from the pool once per batch. Benchmarks are good now

Reply:
awesome. Do you want to file an issue for the jsoniter changes? @crobert-1 can make those if he's interested

Reply:
I'd say we need to check what UF is doing and keep the exporter behavior close to that

Reply:
That's a good conclusion. Let's get this one in and we can table the jsoniter changes until we have more tests against the UF.